functions/Test-DbaAvailabilityGroup.ps1

```diff
@@ -0,0 +1,103 @@
+
+    .PARAMETER SqlCredential
+        Login to the target instance using alternative credentials. Windows and SQL Authentication supported. Accepts credential objects (Get-Credential)
+
+    .PARAMETER AvailabilityGroup
+        Return only specific availability groups.
+
+    .PARAMETER IncludeUserPolicy
+
+        Indicates that this function tests user policies found in the policy categories of Always On Availability Groups.
+
+    .PARAMETER IncludePolicyDetail
+
+        Indicates that this function displays the result of each policy evaluation that it performs. The function returns one object per policy evaluation. Each policy object includes the results of evaluation. This information includes
+        whether the policy passed or not, the policy name, and policy category.
+
+    .PARAMETER InputObject
+
+        Enables piping from Get-DbaAvailabilityGroup
+    
+    .PARAMETER EnableException
+        By default, when something goes wrong we try to catch it, interpret it and give you a friendly warning message.
+        This avoids overwhelming you with "sea of red" exceptions, but is inconvenient because it basically disables advanced scripting.
+        Using this switch turns this "nice by default" feature off and enables you to catch exceptions with your own try/catch.
+
+    .NOTES
+        Tags: Hadr, HA, AG, AvailabilityGroup
+        Author: IJeb Reitsma
+
+        Website: https://dbatools.io
+        Copyright: (c) 2018 by dbatools, licensed under MIT
+        License: MIT https://opensource.org/licenses/MIT
+
+    .LINK
+        https://dbatools.io/Test-DbaAvailabilityGroup
+
+    .EXAMPLE
+
+        PS C:\> Test-DbaAvailabilityGroup -SqlInstance sqlserver-0
+
+        Returns the health state on all the Availability Group(s) found on sqlserver-0.
+
+        PS C:\> Test-DbaAvailabilityGroup -SqlInstance sqlserver-0 -AvailabilityGroup test-ag
+
+        Returns the health state on the Availability Group test-ag on sqlserver-0.
+
+        PS C:\> Test-DbaAvailabilityGroup -SqlInstance sqlserver-0 -AvailabilityGroup test-ag -IncludePolicyDetail
+
+        Returns the health state on the Availability Group test-ag on sqlserver-0. Include detailed results.
+
+        PS C:\> Get-DbaAvailabilityGroup -SqlInstance sqlserver-0 | Test-DbaAvailabilityGroup
+
+        Returns the health state on all the Availability Group(s) found on sqlserver-0.
+
+        PS C:\> Get-DbaAvailabilityGroup -SqlInstance sqlserver-0 -AvailabilityGroup test-ag | Test-DbaAvailabilityGroup
+
+        Returns the health state on the Availability Group test-ag on sqlserver-0.
+
+    #>
+    [CmdletBinding()]
+    param (
+        [DbaInstanceParameter[]]$SqlInstance,
+        [PSCredential]$SqlCredential,
+        [string[]]$AvailabilityGroup,
+        [parameter(ValueFromPipeline)]
+        [Microsoft.SqlServer.Management.Smo.AvailabilityGroup[]]$InputObject,
+        [switch]$IncludeUserPolicy,
```

Comment:
I changed these to singular. I'm surprised that the SqlServer module does plural params.